governance/pyth_staking_sdk/src/pyth-staking-client.ts

```diff
@@ -609,6 +622,24 @@ export class PythStakingClient {
       const buffer = Buffer.from(val, "base64").reverse();
       totalRewards += BigInt("0x" + buffer.toString("hex"));
     }
-    return totalRewards;
+
+    const delegationRecords = await Promise.all(
+      instructions.publishers.map(({ pubkey }) =>
+        this.getDelegationRecord(stakeAccountPositions, pubkey),
+      ),
+    );
+
+    let lowestEpoch: bigint | undefined;
+    for (const record of delegationRecords) {
+      if (lowestEpoch === undefined || record.lastEpoch < lowestEpoch) {
+        lowestEpoch = record.lastEpoch;
+      }
+    }
```

Comment:
@guibescos I'm assuming that when there is no stake (i.e reward is zero) the position will get removed after merging positions. Therefore, this will only account for publishers that we have an stake with.